src/core/thread/mle.cpp

```diff
@@ -259,7 +263,21 @@ void Mle::Stop(StopMode aMode)
     SetRole(kRoleDisabled);
 
 exit:
-    return;
+    if (IsDetachingGracefully())
+    {
+        mDetachGracefullyTimer.Stop();
+    }
```

Comment:
Simplify this?
Stopping an already stopped (not-running) timer is harmless and does nothing.
```suggestion
    mDetachGracefullyTimer.Stop();
```

Reply:
Done.